kubectl-cloudflow/version/version.go

```diff
@@ -30,21 +33,58 @@ const ProtocolVersionKey = "protocol-version"
 // ProtocolVersionConfigMapName is the name of the configmap that contains the protocol-version
 const ProtocolVersionConfigMapName = "cloudflow-protocol-version"
 
-// FailOnProtocolVersionMismatch fails and exits if the protocol version of kubectl-cloudflow does not match with the cloudflow operator protocol version.
-func FailOnProtocolVersionMismatch() {
+// CloudflowNamespaceLabelKey is the key of the label for the cloudflow installation namespace
+const cloudflowNamespaceLabelKey = "namespace"
+
+// CloudflowNamespaceLabelValue is the value of the label for the cloudflow installation namespace
+const cloudflowNamespaceLabelValue = "cloudflow-namespace"
+
+// GetProtocolVersionConfigMap Get the protocol version config map set by the operator
+func GetProtocolVersionConfigMap() (*corev1.ConfigMap, error) {
 	k8sClient, k8sErr := k8s.GetClient()
 	if k8sErr != nil {
 		util.LogAndExit("Failed to create new kubernetes client, %s", k8sErr.Error())
 	}
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{cloudflowNamespaceLabelKey: cloudflowNamespaceLabelValue}}
 
 	var cm *corev1.ConfigMap
-	if configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(metav1.ListOptions{}); err == nil {
-		for i, v := range configMaps.Items {
-			if v.Name == ProtocolVersionConfigMapName {
-				cm = &configMaps.Items[i]
-			}
+	configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(
+		metav1.ListOptions{LabelSelector: labels.Set(labelSelector.MatchLabels).String()})
+	if err == nil {
+		if len(configMaps.Items) > 1 {
+			return nil, errors.New("Multiple cloudflow operators detected in namespace .. exiting")
+		}
+		cm = &configMaps.Items[0]
+		if cm.GetName() == ProtocolVersionConfigMapName {
+			return cm, nil
 		}
-	} else {
+		return nil, fmt.Errorf("Invalid protocol version config %s detected", cm.GetName())
+	}
+	return cm, err
+}
+
+// GetCloudflowNamespace Get the default namespace set in the protocol version config map
+func GetCloudflowNamespace() (string, error) {
```

Comment:
I'm not totally up to scratch on Golang naming conventions but I suggest we rename this to `FindCloudflowNamespace ` since it is not guaranteed that there actually is one. to me it would better describe the purpose and the context at the call site.

Reply:
ok .. will change ..